### DISCUSSION TITLE
crypto-js weakness

### DISCUSSION THREAD

**Discussion by jprtf** [jgraph/drawio#3808] [General]
The `crypto-js` package before `3.2.1` for Node.js generates random numbers by concatenating the string "0." with an integer, which makes the output more predictable than necessary.

The [current version](https://github.com/jgraph/drawio/blob/f819ee37b836313e2a6afefa0b8f50a6dbfcb2ee/etc/dependencies/package.json#L17) in on drawio is `3.1.2`.

Shouldn't this be addressed?

More info:
* https://github.com/advisories/GHSA-3w3w-pxmm-2w2j
* https://nvd.nist.gov/vuln/detail/CVE-2020-36732

**Comment by davidjgraph** [comment]
Thanks for the report. Looking through our usage this doesn't cause a security issue based on these reports. We'll update the library, though, as a matter of course.

**Reply by davidjgraph** [reply]
To follow up, we won't be updating, as we have a custom, stripped down 3.1.2 build that does not contain the random function causing the problem here. So, the manifest will indicate 3.1.2, but that's mostly not the case.

**Reply by jprtf** [reply]
Understood. Thank you David!
